src/dependency.rs

```diff
@@ -14,6 +14,7 @@ pub struct Dependency {
     pub name: String,
     optional: bool,
     source: DependencySource,
+    features: Vec<String>,
```

Comment:
We need to distinguish an empty feature list from the default feature list (not specifying the feature flag).

Reply:
Hi Ordian, 
Thanks for your suggestion. "cargo list" or "cargo read-manifest" can only read the local cargo.toml manifest file. When you add a dependency to your project, you don't save a copy of the manifest file of the dependencies. Without the manifest file of the dependency, there is no way of returning the list of the available features. I think this enhancement is only useful for experienced users who would check out the available features themselves. I am only motivated in using cargo-edit because Cargo does not support Target specific features. https://github.com/rust-lang/cargo/issues/1197
With cargo-edit being able to specify features, I can use continuous integration to build various target builds with dependencies with various features.

Reply:
1. We download `Cargo.toml` [locally](https://github.com/killercup/cargo-edit/blob/master/src/fetch.rs#L287) for git dependencies in cargo-edit (and download json for crates.io dependencies), so a proper way to extract feature list would be to create a function `get_features_from_manifest` similar to `get_name_from_manifest`. But that can be done in a separate PR.
2. The point of the comment is different: if user of cargo-edit does not specify `--features` when adding a dependency, he probably wants a [default feature list](https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section), not the empty one. Btw, [current CI failure](https://travis-ci.org/killercup/cargo-edit/jobs/334308301#L1053-L1055) is relevant.